sdk/identity/Azure.Identity/src/MsalConfidentialClient.cs

```diff
@@ -75,7 +76,7 @@ protected override async ValueTask<IConfidentialClientApplication> CreateClientA
             if (_appTokenProviderCallback != null)
             {
                 confClientBuilder.WithAppTokenProvider(_appTokenProviderCallback)
-                    .WithAuthority(_authority.AbsoluteUri, TenantId, false)
+                    .WithAuthority("https://aka.ms/azsdk/net/identity/managedidentitycredential/troubleshoot", TenantId, false)
```

Comment:
Have you tried using the new option to completely disable instance metadata? We tested that it plays well with regional. `WithInstanceDiscovery(false)`